src/coreclr/jit/llvmlssa.cpp

```diff
@@ -642,6 +642,67 @@ unsigned Llvm::getShadowFrameSize(unsigned hndIndex) const
     return 0;
 }
 
+ValueInitKind Llvm::getInitKindForLocal(unsigned lclNum) const
+{
+    LclVarDsc* varDsc = _compiler->lvaGetDesc(lclNum);
+
+    // Is the value live on entry?
+    if (varDsc->lvHasExplicitInit)
+    {
+        // No - no need to initialize it, even in the GC case.
+        return ValueInitKind::None;
+    }
+
+    // We choose to always initialize GC values to reduce the number of "random" pointers on the shadow stack.
```

Comment:
This is a costly policy and one we _may_ want to reconsider. The cost is around `1.5%` in code size for HelloWasm.

Two fixes are possible. The simple one is to just omit the zeroing for values not explicitly live. Another is to have precise GC.

Optimizations are also possible. E. g. do not zero in methods without safepoints. Do not zero if the value (a shadow stack slot, really) is not "implicitly" live across a safepoint. Certainly something to keep in mind for #2294.